corporate/lib/stripe.py

```diff
@@ -1919,6 +1919,10 @@ def do_update_plan(self, update_plan_request: UpdatePlanRequest) -> None:
 
         licenses = update_plan_request.licenses
         if licenses is not None:
+            if plan.is_free_trial():
+                raise JsonableError(
+                    _("Cannot update licenses in the current billing period for free trial plan.")
+                )
             if plan.automanage_licenses:
```

Comment:
Should we add a check for `downgrade_at_end_of_free_trial` and `downgrade_at_end_of_cycle` for `licenses_at_next_renewal` too? I wasn't sure why we don't have one for `downgrade_at_end_of_cycle`.

Reply:
Added the checks.